collection/src/collection.rs

```diff
@@ -83,7 +88,7 @@ pub struct Collection<T> {
 /// # Examples
 ///
 /// ```
-/// # use transit_model::collection::*;
```

Comment:
Technically, these tests should be considered like integration tests, as a user of the final packaged crate. Especially, it would be awkward to have a documentation that tells me
```
use crate::collection::*;
```
I know all of the modifications you made concerns invisible documentation but yet I feel that keeping `transit_model` is more correct.

Reply:
I don't understand your point. This PR is to make collection indenpendant of transit_model

Reply:
First of all, I misread something and indeed, I didn't meant `transit_model` but `collection`. So basically, having `use collection::collection::*` instead of `use crate::collection::*`.

Another way of seeing this is: documentation is supposed to contain code examples you should be able to copy/paste and build. If you put something like `crate::*` in the documentation, then whoever is gonna copy/paste that is gonna need to figure out what `crate` means in her/his context.

Code example of the documentation should be written as user of the final package. A user of the final package doesn't have access to `crate` but to `collection`.

Reply:
Maybe the `crate::Result` too could be transformed into `collection::Result`.